crates/storage/src/pack.rs

```diff
@@ -79,6 +126,25 @@ impl<T> Pack<T> {
     pub fn as_inner_mut(pack: &mut Pack<T>) -> &mut T {
         &mut pack.inner
     }
+
+    /// Creates a new packed value with a `key`.
```

Comment:
Please add comment to describe that this is a low-level API specifically designed for `spread_pull`. However, I still kinda don't really see its point besides perhaps some code dedupe.